src/erl/typespec.rs

```diff
@@ -0,0 +1,263 @@
+use crate::{ast::*, erl::format::*, erl::literals::atom, erl::names, pretty::*, typ::Type};
+
+use heck::{CamelCase, SnakeCase};
+use itertools::Itertools;
+
+// Creates the contents of the `export_type([..])` line
+pub fn type_exports(module: &TypedModule) -> Document {
+    concat(
+        module
+            .statements
+            .iter()
+            .flat_map(|s| match s {
+                Statement::CustomType {
+                    public: true,
+                    name,
+                    constructors,
+                    ..
+                } => {
+                    // We only count the params that we can find in the constructors as we can only
+                    // put those ones into the type definition as otherwise we'll have unused type
+                    // param warnings
```

Comment:
I'm not sure what this means?

Reply:
Good question, the clearest example would be:
```
pub external fn from(a) -> Dynamic =
  "gleam_stdlib" "identity"
```
If we generate:
```erlang
-spec from(A) -> dynamic.
```
Then we get an error saying that A is unused I believe. Instead we have to say:
```erlang
-spec from(any) -> dynamic.
```
And we can figure out which type parameters to change to `any`s by inspecting (in this case) the return type and figuring out how many type parameters are used in it and replacing any others with `any` in the spec. 

I came across this with types too. Possibly just due to lazy type generation. I don't have code that handled function types for example so if a type parameter is used in a function type in a type definition then it isn't going to be output so we don't want to include it initially. I realise that "laziness" is not a good excuse here but it was done in order to get something imperfect out to see if the direction made sense rather than trying to work through all the possible cases.